src/Renderer/ThreeExtended/PlanarControls.js

```diff
@@ -0,0 +1,762 @@
+
+    // max travel duration is reached for this travel distance
+    this.autoTravelTimeDist = options.autoTravelTimeDist || 20000;
+
+    // after a smartZoom, camera height above ground will be between these two values
+    this.smartZoomHeightMin = options.smartZoomHeightMin || 75;
+    this.smartZoomHeightMax = options.smartZoomHeightMax || 500;
+
+    // if set to true, animated travels have 0 duration
+    this.instantTravel = options.instantTravel || false;
+
+    this.minZenithAngle = options.minZenithAngle || 0 * Math.PI / 180;
+
+    // should be less than 90 deg (90 = parallel to the ground)
+    this.maxZenithAngle = (options.maxZenithAngle || 82.5) * Math.PI / 180;
+
+    // prevent the default contextmenu from appearing when right-clicking
+    // this allows to use right-click for input without the menu appearing
+    this.domElement.addEventListener('contextmenu', onContextMenu.bind(this), false);
+
+    // add this PlanarControl instance to the view's framerequesters
+    // with this, PlanarControl.update() will be called each frame
+    this.view.addFrameRequester(this);
+
+    _handlerOnKeyDown = onKeyDown.bind(this);
+    _handlerOnKeyUp = onKeyUp.bind(this);
+    _handlerOnMouseDown = onMouseDown.bind(this);
+    _handlerOnMouseUp = onMouseUp.bind(this);
+    _handlerOnMouseMove = onMouseMove.bind(this);
+    _handlerOnMouseWheel = onMouseWheel.bind(this);
+
+    /** =========================================================================================================================
+    * PlanarControl update
+    * Updates the view and camera if needed, and handles the animated travel
+    */
+    this.update = function update() {
+        deltaTime = clock.getElapsedTime() - lastElapsedTime;
+        lastElapsedTime = clock.getElapsedTime();
+
+        if (state === STATE.TRAVEL) {
+            this.handleTravel(deltaTime);
+        }
+
+        if (state !== STATE.NONE) {
+            this.view.camera.update(window.innerWidth, window.innerHeight);
+
+            this.view.notifyChange(true);
+        }
+    };
+
+    /** =========================================================================================================================
+    * Initiate a drag movement (translation on xy plane) when user does a left-click
+    * The movement value is derived from the actual world point under the mouse cursor
+    * This allows the user to 'grab' a world point and drag it to move (eg : google map)
+    */
+    this.initiateDrag = function initiateDrag() {
+        state = STATE.DRAG;
+
+        // the world point under mouse cursor when the drag movement is started
+        dragStart.copy(this.getWorldPointAtScreenXY(mousePosition));
+
+        // the difference between start and end cursor position
+        dragDelta.set(0, 0, 0);
+    };
+
+    /** =========================================================================================================================
```

Comment:
I don't mind the `=========` separator but this is something that should be done everywhere or nowhere (as all other stylistic choices).
So I'd suggest removing them from this PR and open a specific PR to convert itowns to this style (but I'm not sure it would be merged...)